selection/algorithms/epic_algorithm.py

```diff
@@ -89,18 +95,28 @@ def _get_candidates_within_budget(self, index_combination_size,
                 new_candidates.append(candidate)
         return new_candidates
 
-    def _evaluate_combination(self, index_combination, best, current_cost):
+    def _evaluate_combination(self,
+                              index_combination,
+                              best,
+                              current_cost,
+                              old_index_size=0):
```

Comment:
I think we do not need `old_index_size` if we store and compare the improvement based on the initial costs.
I also think that the code would then be easier to understand

Reply:
I agree regarding the understandability but this version performs better and is closer to the original version of the algorithm presented in the 2019 ICDE paper.